### ISSUE TITLE
Support `class_weight` for TimeSeries models in Keras (`model.fit`)

### ISSUE THREAD

**Issue by itsnamgyu** [tensorflow/tensorflow#45649]
**System information**
- TensorFlow version (you are using): 2.3
- Are you willing to contribute it (Yes/No): As much as I can

**Describe the feature and the current behavior/state.**
Currently it is only possible to specify `class_weight` in `tf.keras.engine.training.Model.fit` for y (labels) with at most 2 dimensions `(batch, class)`. Would it be possible to support weights for TimeSeries classification models with three-dimensional labels `(batch, time_step, class)`?

Note that it is currently possible to specify timestep-specific weights for individual input samples using `sample_weights`.

**Will this change the current api? How?**
Maybe. Currently, `class_weight` is accepted in the form of `Dict[int, float]`. It may be possible to change the format to `np.array` with 1 or 2 dimensions. It is also possible to maintain the current argument format and apply the same class weights to each time-step uniformly 

**Who will benefit with this feature?**
People who are training time-series classification models with imbalanced classes.

**Any Other info.**
n/a

**Comment by bhack** [comment]
Take a look at https://github.com/keras-team/keras/issues/3653
